azure/iac/Redis.rego

```diff
@@ -142,4 +142,208 @@ serverRole_metadata := {
+    not azure_issue["redis_public_access"]
+}
+
+redis_public_access = false {
+    azure_attribute_absence["redis_public_access"]
+}
+
+redis_public_access = false {
+    azure_issue["redis_public_access"]
+}
+
+redis_public_access_err = "Azure Redis Cache with public access detected!" {
+    azure_issue["redis_public_access"]
+} else = "Azure Cache for Redis attribute publicNetworkAccess missing in the resource" {
+    azure_attribute_absence["redis_public_access"]
+}
+
+redis_public_access_metadata := {
+    "Policy Code": "PR-AZR-ARM-ARC-003",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Azure Cache for Redis should disable public network access",
+    "Policy Description": "Disabling public network access improves security by ensuring that the Azure Cache for Redis isn't exposed on the public internet. You can limit exposure of your Azure Cache for Redis by creating private endpoints instead. Learn more at: https://docs.microsoft.com/azure/azure-cache-for-redis/cache-private-link.",
+    "Resource Type": "microsoft.cache/redis",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redis"
+}
+
+
+# PR-AZR-ARM-ARC-004
+#
+
+default arc_subnet_id = null
+
+azure_attribute_absence["arc_subnet_id"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    not resource.properties.subnetId
+}
+
+source_path[{"arc_subnet_id":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.cache/redis"
+    not resource.properties.subnetId
+    metadata:= {
+        "resource_path": [["resources",i,"properties","subnetId"]]
+    }
+}
+
+
+azure_issue["arc_subnet_id"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    count(resource.properties.subnetId) == 0
+}
+
+
+source_path[{"arc_subnet_id":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.cache/redis"
+    count(resource.properties.subnetId) == 0
+    metadata:= {
+        "resource_path": [["resources",i,"properties","subnetId"]]
+    }
+}
+
+arc_subnet_id {
+    lower(input.resources[_].type) == "microsoft.cache/redis"
+    not azure_issue["arc_subnet_id"]
+    not azure_attribute_absence["arc_subnet_id"]
+}
+
+arc_subnet_id = false {
+    azure_issue["arc_subnet_id"]
+}
+
+arc_subnet_id = false {
+    azure_attribute_absence["arc_subnet_id"]
+}
+
+arc_subnet_id_err = "Azure Cache for Redis is not reside within a virtual network" {
+    azure_issue["arc_subnet_id"]
+} else = "Azure Cache for Redis attribute subnetId missing in the resource" {
+    azure_attribute_absence["arc_subnet_id"]
+}
+
+
+arc_subnet_id_metadata := {
+    "Policy Code": "PR-AZR-ARM-ARC-004",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Azure Cache for Redis should use private link",
```

Comment:
@shahinshirvani title seems same with `PR-AZR-ARM-ARC-005`
Title should be 'Azure Cache for Redis should reside within a virtual network'